controller-server/src/main/java/com/yahoo/vespa/hosted/controller/versions/VersionStatus.java

```diff
@@ -130,6 +130,9 @@ public static VersionStatus compute(Controller controller) {
         ListMap<Version, HostName> versions = new ListMap<>();
         for (ZoneId zone : zones) {
             for (SystemApplication application : SystemApplication.all()) {
+                if (!application.configConvergedIn(zone, controller)) {
+                    throw new IllegalStateException("Config for " + application.id() + " in " + zone + " has not converged");
```

Comment:
I haven't checked the implementation of configConverged, but won't this often be (`!`) false during normal operations as well? This code is run regularly.

Reply:
I guess the question is: how likely is this condition to be false, at any randomly chosen point in time?

Reply:
I guess it's fine as long as this probability is not high.

Reply:
It's unlikely, but it may coincide with internal redeployment (which is every 30 minutes). If that happens it will just keep the current system version, which is fine, and retry on next run.

I think the chance of colliding is low enough to not be an issue.